pkg/api/controller_test.go

```diff
@@ -1285,3 +1285,95 @@ func TestController_SetupLakeFSHandler(t *testing.T) {
 		})
 	}
 }
+
+func TestController_ListRepositoryRuns(t *testing.T) {
+	clt, _ := setupClientWithAdmin(t, "")
+	ctx := context.Background()
+	httpServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusOK)
+	}))
+	defer httpServer.Close()
+	// create repository
+	resp, err := clt.CreateRepositoryWithResponse(ctx, &api.CreateRepositoryParams{}, api.CreateRepositoryJSONRequestBody{
+		DefaultBranch:    api.StringPtr("master"),
+		Name:             "repo9",
+		StorageNamespace: "mem://repo9",
+	})
+	verifyResponseOK(t, resp, err)
+	// upload action for pre-commit
+	actionContent := `--- 
+name: CommitAction
+on:
+  pre-commit:
+    branches:
+      - "*"
+hooks:
+  - id: hook1
+    type: webhook
+    properties:
+      url: ` + httpServer.URL
+	uploadResp, err := uploadObjectHelper(t, ctx, clt, "_lakefs_actions/pre_commit.yaml", strings.NewReader(actionContent), "repo9", "master")
+	verifyResponseOK(t, uploadResp, err)
+	// commit
+	respCommit, err := clt.CommitWithResponse(ctx, "repo9", "master", api.CommitJSONRequestBody{
+		Message: "pre-commit action",
+	})
+	verifyResponseOK(t, respCommit, err)
+	// work branch
+	branchResp, err := clt.CreateBranchWithResponse(ctx, "repo9", api.CreateBranchJSONRequestBody{
+		Name:   "work",
+		Source: "master",
+	})
+	verifyResponseOK(t, branchResp, err)
+	// upload and commit content on branch
+	commitIDs := []string{respCommit.JSON201.Id}
+	const contentCount = 5
+	for i := 0; i < contentCount; i++ {
+		content := fmt.Sprintf("content-%d", i)
+		uploadResp, err := uploadObjectHelper(t, ctx, clt, content, strings.NewReader(content), "repo9", "work")
+		verifyResponseOK(t, uploadResp, err)
+		respCommit, err := clt.CommitWithResponse(ctx, "repo9", "work", api.CommitJSONRequestBody{Message: content})
+		verifyResponseOK(t, respCommit, err)
+		commitIDs = append(commitIDs, respCommit.JSON201.Id)
+	}
+
+	t.Run("total", func(t *testing.T) {
+		respList, err := clt.ListRepositoryRunsWithResponse(ctx, "repo9", &api.ListRepositoryRunsParams{
+			Amount: api.PaginationAmountPtr(100),
+		})
+		verifyResponseOK(t, respList, err)
+		runsCount := len(respList.JSON200.Results)
+		expectedRunsCount := contentCount + 1 // additional one for the actions commit
+		if runsCount != expectedRunsCount {
+			t.Fatalf("ListRepositoryRuns() got %d results, expected %d", runsCount, expectedRunsCount)
+		}
```

Comment:
I think this error message will be easier to read.  Also the code is shorter.
```suggestion
		if runsCount != contentCount + 1 { // additional one for the actions commit
			t.Fatalf("ListRepositoryRuns() got %d results, expected %d+1", runsCount, contentCount+1)
		}
```